### PR TITLE
complete NAPI_Initialize NAPI_Finalize test_AES128 

### PR DESCRIPTION
**Pull request by Liang1XZhao**
change function name parseStringData -> readData 
in the ehsm_kms_service initialize and finalize

### CODE REVIEW

**Review by syan10 — APPROVED** [review]

File: ehsm_kms_service/ehsm_kms_server.js [hunk 1]
```diff
@@ -186,8 +186,20 @@ const ehsm_napi = ffi.Library('./libehsmnapi',{
...
+  'NAPI_Initialize':['string',[]],
+  
+  /*
+  destory the enclave
+  */
+  'NAPI_Finalize':['string', []],
+
 });
 
+ehsm_napi.NAPI_Initialize();
```
Comment (yang8621) [inline ehsm_kms_service/ehsm_kms_server.js:202]: Check the result

File: core/App/ehsm_napi.cpp [hunk 2]
```diff
@@ -36,53 +36,37 @@
...
-    char *retChar = NULL;
-    if (str.size() > 0) {
-        int len = str.size();
-        retChar = (char *)malloc(len * sizeof(uint8_t));
-        if(retChar != nullptr){
-            memset(retChar, 0, len);
-            memcpy(retChar, str.c_str(), len);
-        }
+
+extern "C" {
```
Comment (yang8621) [inline core/App/ehsm_napi.cpp:40]: Is it really required?

File: core/App/ehsm_napi.h [hunk 3]
```diff
@@ -36,15 +36,84 @@ using namespace std;
 
 extern "C" {
```
Comment (yang8621) [inline core/App/ehsm_napi.h:37]: Is it really required?

File: core/App/ehsm_napi.cpp (hunk 2 above)
Comment (285241) [reply core/App/ehsm_napi.cpp:40]: It will be remove on next PR.

File: core/App/ehsm_napi.h (hunk 3 above)
Comment (285241) [reply core/App/ehsm_napi.h:37]: It will be remove on next PR.

File: core/App/ehsm_napi.cpp (hunk 2 above)
Comment (285241) [reply core/App/ehsm_napi.cpp:40]: It is  really required,for nodejs.

File: core/App/ehsm_napi.h (hunk 3 above)
Comment (285241) [reply core/App/ehsm_napi.h:37]: It is  really required,for nodejs.

File: core/App/ehsm_napi.cpp (hunk 2 above)
Comment (Liang1XZhao) [reply core/App/ehsm_napi.cpp:40]: yes,for nodejs.

File: core/App/ehsm_napi.h (hunk 3 above)
Comment (Liang1XZhao) [reply core/App/ehsm_napi.h:37]: yes,for nodejs.

**Comment by Liang1XZhao** [comment]
Resubmit after repair

**Pull request closed without merging**